src/main/java/org/jenkinsci/plugins/workflow/cps/CpsGroovyShell.java

```diff
@@ -110,7 +110,7 @@ public void prepareScript(Script script) {
     public Script parse(GroovyCodeSource codeSource) throws CompilationFailedException {
         Script s = doParse(codeSource);
         if (execution!=null)
-            execution.loadedScripts.put(s.getClass().getName(), codeSource.getScriptText());
+            execution.loadedScripts.put(s.getClass().getSimpleName(), codeSource.getScriptText());
```

Comment:
It is a user error to include a `package` declaration in a `load`ed script, so maybe we should just throw an explanatory error right here.

Reply:
Warning at least would make sense -- but that need not be part of this PR.

Reply:
Yes a warning would be a good idea. Could be in this PR but does not need to be.

Reply:
Yeah, I'm thinking of doing that in another PR - I'm debating whether the text should be just a pure "don't do that" warning or something more like "We're gonna error on this eventually, so *really* don't do that" warning, and I want to ponder that for a bit. =)